docker/php-fpm/Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM php:8.2-fpm
+
+COPY wait-for-it.sh /usr/bin/wait-for-it
+
+RUN chmod +x /usr/bin/wait-for-it
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends libssl-dev zlib1g-dev curl git unzip netcat libxml2-dev libpq-dev libzip-dev && \
+    pecl install apcu && \
+    docker-php-ext-configure pgsql -with-pgsql=/usr/local/pgsql && \
+    docker-php-ext-install -j$(nproc) zip opcache intl pdo_pgsql pgsql && \
+    docker-php-ext-enable apcu pdo_pgsql sodium && \
+    apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+COPY --from=composer /usr/bin/composer /usr/bin/composer
+
+WORKDIR /var/www
+
+CMD composer i -o; wait-for-it db:5432 -- bin/console doctrine:migrations:migrate; php-fpm
```

Comment:
Le wait for db devrait être fait dans le docker-compose également. Normalement un docker file est sensé être indépendant, car il a pour but de démarrer un conteneur avec une philosophie d'isolation. Ici du coup, tu bloques son démarrage s'il n'y a aucune base de données de branchée.
De plus, le port est ultra-spécifique à la plateforme utilisée.

Une approche native, et plutôt moderne serait d'utiliser le health checker de docker-compose (regarde cet article à la partie health check) https://laurent-bel.medium.com/waiting-for-postgresql-to-start-in-docker-compose-c72271b3c74a